### PR TITLE
Github Fine-granted Token pattern

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- make the package PEP-561 compliant (added [py.typed](https://peps.python.org/pep-0561/) file)
- Github Fine-granted Token pattern (https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/)
- added test samples for each pattern rule of regex type

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] Import the package to a project and run mypy analyzer - should be no errors about importing types.

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ def test_binary_patch_n(self, mock_get_arguments: Mock()) -> None:
             app_main.main()
             # two times when analysis passed "added data" + two in "deleted data" case
             mocked_logger.assert_called()
-            assert mocked_logger.call_count == 4
+            # two binary files in the diff
+            assert mocked_logger.call_count == 2
```
Comment (babenek) [inline tests/test_main.py:174]:
```suggestion
            # two binary files in the diff. whatthepatch 1.0.3 produces more None diffs
            assert mocked_logger.call_count >= 2
```

File: credsweeper/rules/config.yaml [hunk 2]
```diff
@@ -424,7 +424,7 @@
   severity: high
   type: pattern
   values:
-    - (?P<value>[0-9]{10}:AA[\w\\-_-]{33})
+    - (?P<value>[0-9]{8,10}:[0-9A-Za-z_-]{35})
```
Comment (csh519) [inline credsweeper/rules/config.yaml:427]: Is `Telegram Bot API Token` has been changed?
If it is changed, this change should be included in other PR.
Please check.

File: tests/test_app.py [hunk 3]
```diff
@@ -580,3 +580,18 @@ def test_denylist_line_n(self) -> None:
...
                 report = json.load(json_file)
                 assert len(report) == 1
+
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+    def test_rules_p(self) -> None:
+        _stdout, _stderr = self._m_credsweeper(["--log", "silence", "--ml_threshold", "0", "--path", str(SAMPLES_DIR)])
+        assert len(_stderr) == 0
+        output = _stdout.decode(errors='replace')
+        # print(output)
```
Comment (csh519) [inline tests/test_app.py:590]: Annotated line should be removed :)

File: tests/test_app.py [hunk 4]
```diff
@@ -580,3 +580,18 @@ def test_denylist_line_n(self) -> None:
...
+        rules = Util.yaml_load(PROJECT_DIR / "credsweeper" / "rules" / "config.yaml")
+        for rule in rules:
+            rule_name = rule["name"]
+            if rule_name in ["Nonce", "Salt", "Certificate"]:
+                print(f" Skip: '{rule_name}'")
```
Comment (csh519) [inline tests/test_app.py:595]: This `print()` method seems like for debug.. if it is true, could you please remove this line for clean code?

File: credsweeper/rules/config.yaml (hunk 2 above)
Comment (babenek) [reply credsweeper/rules/config.yaml:427]: There are many samples of the token type, but not any rule.
https://core.telegram.org/bots/api#authorizing-your-bot
https://core.telegram.org/bots/features#creating-a-new-bot
I think, left numeric side is grown and now it is 10 symbols, but oldest tokens might be with 9 or less symbols (as it is shown in examples)
Current pattern id matched extra charachters ``^`` and ``]``  due the seuqence ``\\-_`` 
I propose to fix it in the PR:
``- (?P<value>[0-9]{10}:AA[\w-]{33})``

File: tests/test_app.py (hunk 3 above)
Comment (babenek) [inline tests/test_app.py:590]:
```suggestion
```

File: tests/test_app.py (hunk 4 above)
Comment (babenek) [inline tests/test_app.py:595]:
```suggestion
```

**Review by babenek — COMMENTED** [review]
fix issues

File: credsweeper/rules/config.yaml (hunk 2 above)
Comment (csh519) [reply credsweeper/rules/config.yaml:427]: Okay.
Let's fix the pattern.

**Review by csh519 — APPROVED** [review]
Telegram pattern changed and new Github Fine-granted Token added.
Also more samples for tests are added.

LGTM :+1:

**Review by kmnls — APPROVED** [review]
Agree, pattern is stricter now

**Review by iuriimet — APPROVED** [review]
Looks good

**Pull request merged**